src/weapon.py

```diff
@@ -56,9 +56,7 @@ def __init__(
     ):
         super().__init__(input_button, car, weapon_sprite_offset)
         self.beam_range = 500
-        self.beam_projection = arcade.SpriteSolidColor(
-            self.beam_range, 5, arcade.color.RED
-        )
+        self.beam_projection: SpriteForBeam = SpriteForBeam(self)
```

Comment:
`SpriteForBeam` was already here from another PR, but not being used?

Reply:
I merged Liam's code with the `SpriteForBeam` functionality into this, and I think it's being used right. Let me know if I'm missing something

Reply:
If something isn't implemented yet, it's good practice to have that be explicit in the code.  If it doesn't break the game, I think it should raise a NotImplementedError; or if it does, log to the console that it's not implemented yet.

Reply:
Sorry for the confusion, `SpriteForBeam` is implemented. Andrew was just surprised to see it since it was from Liam's change order that I merged in to my branch from the updated main to avoid conflicts. I read too much into his comment

Reply:
In this case, `SpriteForBeam` is fully implemented, and is being used, so raising an error doesn't fit.  The collision detection logic that would rely on it isn't implemented, so the motivation for this class's existence is unclear to someone naively reading the code.  Dan and Liam have time scheduled on Sunday to work on that bit.  For the time being, I personally think it's fine to keep this class in the codebase.

All these `SpriteFor*` classes will likely be generalized into a single, generic type.  At least that's my prediction.